### PR TITLE
Export all convert related functions from core

### PR DESCRIPTION
**Pull request by milesj**
@kentcdodds 

**What**: Moves all converter functions to an internal file so that they could be exported and easily used from core.

**Why**: I consistently find myself wanting to use `convertProperty`, as it's much easier to use when you have key/value variables and want to avoid creating a temporary object.

**How**: Exporting `convertProperty` from the index would cause "default" issues, so I had to move it to another file. Moving it also required everything else to be moved as they are circularly referenced.

**Checklist**:

- [ ] Documentation N/A
- [x] Tests
- [x] Ready to be merged 
- [ ] Added myself to contributors table

### CODE REVIEW

File: core.d.ts [hunk 1]
```diff
@@ -1,73 +1,85 @@
+export declare type PropertyValue = string | number | object
+
 export declare interface CommonConverterOptions {
-  value: string;
+  value: string
```
Comment (milesj) [inline core.d.ts]: Seems like the `kcd format` is making these changes.

**Comment by kentcdodds** [comment]
This will need to be rebased now.

**Comment by milesj** [comment]
Will do! I'll dig around and see why this coverage isn't working either.

**Comment by milesj** [comment]
@kentcdodds Rebased. The coverage for core wasn't working because nothing was importing it anymore, so I updated one of the tests to import from core instead of the index and now everything has 100% again.

**Comment by milesj** [comment]
Thanks for merging!

**Pull request merged**